File: EcuacionPrimerGrado.php
<?php

$a = readline("Ingrese a: ");
$b = readline("Ingrese b: ");



if ($a == 0 && $b == 0) {
    echo "Infinita solución";
} 
elseif ($a == 0){
    echo "Sin solución";
} 
else {
    $x = 0 - $b / $a;
    echo "Solución única:". $x;
}

File: EcuacionPrimerGrado.py

a = int(input("Ingrese a: "))
b = int(input("Ingrese b: "))

if  a == 0 and b == 0:
    print("Infinita solución") 
elif a == 0:
    print("Sin solución")
else :
    x = 0 - b / a
    print("Solución única:", x) 